test/object-store/migrations.cpp

```diff
@@ -50,8 +50,8 @@ using util::any_cast;
     do {                                                                                                             \
         REQUIRE_NOTHROW((r).update_schema(s, version));                                                              \
         VERIFY_SCHEMA(r, false);                                                                                     \
-        REQUIRE((r).schema() == s);                                                                                  \
```

Comment:
This is a very important check for these tests.

Reply:
OK because the schema is always going to be equal to OS schema. This `REQUIRE` does not apply any more, unless it is a problem to return the OS schema.

Reply:
I replaced this check with a for loop, in which I go through the migrate schema objects and I search for them within the returned schema

Reply:
This is not fixed. The purpose of this assertion is to verify that the reported schema is correct after a migration, and it's no longer doing that.

Reply:
In additive mode, this will never be the case.

Reply:
It is doing that, just not for additive mode. In case the schema is additive, the constraint is relaxed (this is the point of merging the 2 schemas, right?) ... I am actually verifying that the migrated schema is contained in the OS one.

Reply:
The schemas should still be equal for most of the additive mode tests. Only the tests which are specifically testing scenarios where the schemas are expected to be different should skip checking if they are equal.

Reply:
OK I just changed my tests that are aiming at verifying the additive schema behaviour and reverted the changes to the macro.

Reply:
Macro has been reverted to the original state and specific checks put in place for the additive case.